src/main/java/net/minecraftforge/items/wrapper/ShulkerItemStackInvWrapper.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.core.Direction;
+import net.minecraft.core.NonNullList;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.ContainerHelper;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemHandlerHelper;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+@Mod.EventBusSubscriber(modid = "forge")
+public class ShulkerItemStackInvWrapper implements IItemHandler, ICapabilityProvider
```

Comment:
Implementing `ICapabilityProvider` here makes this public API while it should be an implementation detail.

Reply:
Additionally, `IItemHandlerModifiable` should be implemented as well.

Reply:
Why should it be different than any of the other item stack capability wrappers that Forge providers? Considering that every item stack capability wrapper already follows this pattern, including [`FluidBucketWrapper`](https://github.com/MinecraftForge/MinecraftForge/blob/c2225ab1da58cf8eb3ac3aab1aab73f2d7a696fa/src/main/java/net/minecraftforge/fluids/capability/wrappers/FluidBucketWrapper.java#L32), [`FluidHandlerItemStack`](https://github.com/MinecraftForge/MinecraftForge/blob/c2225ab1da58cf8eb3ac3aab1aab73f2d7a696fa/src/main/java/net/minecraftforge/fluids/capability/templates/FluidHandlerItemStack.java#L30), and [`FluidHandlerItemStackSimple`](https://github.com/MinecraftForge/MinecraftForge/blob/c2225ab1da58cf8eb3ac3aab1aab73f2d7a696fa/src/main/java/net/minecraftforge/fluids/capability/templates/FluidHandlerItemStackSimple.java#L27).